src/app_service/service/message/service.nim

```diff
@@ -542,4 +553,21 @@ proc getRenderedText*(self: Service, parsedTextArray: seq[ParsedText]): string =
           "</table>")
       of PARSED_TEXT_TYPE_CODEBLOCK:
         result = result & "<code>" & escape_html(parsedText.literal) & "</code>"
-    result = result.strip()
\ No newline at end of file
+    result = result.strip()
+
+proc deleteMessage*(self: Service, messageId: string) =
+  try:
+    let response = status_go.deleteMessageAndSend(messageId)
+
+    var deletesMessagesObj: JsonNode
+    if(response.result.getProp("removedMessages", deletesMessagesObj)):
+      let deletedMessagesArr = deletesMessagesObj.getElems()
+      if(deletedMessagesArr.len > 0): # an array is returned
+        let deletedMessageObj = deletedMessagesArr[0]
+        var chat_Id, message_Id: string
+        if deletedMessageObj.getProp("chatId", chat_Id) and deletedMessageObj.getProp("messageId", message_Id):
+          let data = MessageDeletedArgs(chatId: chat_Id, messageId: message_Id)
+          self.events.emit(SIGNAL_MESSAGE_DELETION, data)
```

Comment:
@Khushboo-dev-cpp this is ok, and it works, but in order to easier debug things and get notified when something gets wrong I rather prefer other approach here that we know in each point what is happening something like this:
```
proc deleteMessage*(self: Service, messageId: string) =
    try:
      let response = status_go.deleteMessageAndSend(messageId)

      var deletesMessagesObj: JsonNode
      if(not response.result.getProp("removedMessages", deletesMessagesObj) or deletesMessagesObj.kind != JArray):
        error "error: ", methodName="deleteMessage", errDesription = "there is no expected removed message array or it's not an array"
        return

      let deletedMessagesArr = deletesMessagesObj.getElems()
      if(deletedMessagesArr.len == 0): # an array is returned
        error "error: ", methodName="deleteMessage", errDesription = "returned removed message array is empty"
        return

      let deletedMessageObj = deletedMessagesArr[0]
      var chat_Id, message_Id: string
      if not deletedMessageObj.getProp("chatId", chat_Id) or 
        not deletedMessageObj.getProp("messageId", message_Id):
          error "error: ", methodName="deleteMessage", errDesription = "there is no set chat id or message id in response"
          return
      
      let data = MessageDeletedArgs(chatId: chat_Id, messageId: message_Id)
      self.events.emit(SIGNAL_MESSAGE_DELETION, data)

    except Exception as e:
      error "error: ", methodName="deleteMessage", errName = e.name, errDesription = e.msg
```

Reply:
sure will update this :)